File: php/examples/notifications/v1/CreateNotificationSubscriptionSample.php
<?php
/**
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

require_once __DIR__ . '/../../../vendor/autoload.php';
require_once __DIR__ . '/../../Authentication/Authentication.php';
require_once __DIR__ . '/../../Authentication/Config.php';
// [START merchantapi_create_notification_subscription]
use Google\ApiCore\ApiException;
use Google\Shopping\Merchant\Notifications\V1\Client\NotificationsApiServiceClient;
use Google\Shopping\Merchant\Notifications\V1\CreateNotificationSubscriptionRequest;
use Google\Shopping\Merchant\Notifications\V1\NotificationSubscription;
use Google\Shopping\Merchant\Notifications\V1\NotificationSubscription\NotificationEventType;

/**
 * Creates a Notification Subscription.
 */
class CreateNotificationSubscription
{
    /**
     * Helper function to construct the parent resource name.
     *
     * @param string $accountId
     *  The Merchant Center account ID.
     * @return string
     *  The parent resource name.
     */
    private static function getParent($accountId)
    {
        return "accounts/" . $accountId;
    }

    /**
     * Creates a Notification Subscription for a given Merchant Center account.
     *
     * @param array $config
     *  Configuration data for authentication and account details.
     * @param string $callbackUri
     *  The URI to send notifications to.
     * @param int $eventType
     *  The type of event to subscribe to.
     * @throws ApiException
     */
    public static function createNotificationSubscriptionSample($config, $callbackUri, $eventType): void
    {
        // Get OAuth credentials.
        $credentials = Authentication::useServiceAccountOrTokenFile();

        // Set up client options.
        $options = ['credentials' => $credentials];

        // Create a client instance.
        $notificationsApiServiceClient = new NotificationsApiServiceClient($options);

        // Construct the parent resource name.
        $parent = self::getParent($config['accountId']);

        // Create the NotificationSubscription object.
        $notificationSubscription = new NotificationSubscription([
            // Uncomment to apply to all managed accounts.
            // 'all_managed_accounts' => true,
            'target_account' => $parent,
            'call_back_uri' => $callbackUri,
            'registered_event' => $eventType,
        ]);

        // Create the request object.
        $request = new CreateNotificationSubscriptionRequest([
            'parent' => $parent,
            'notification_subscription' => $notificationSubscription,
        ]);

        print "Sending create Notification Subscription request:\n";

        // Make the API call.
        try {
            $response = $notificationsApiServiceClient->createNotificationSubscription($request);
            print "Created Notification Subscription below:\n";
            print_r($response);
        } catch (ApiException $e) {
            print "Request failed:\n";
            print $e->getMessage() . "\n";
        }
    }

    /**
     * Execute the sample.
     *
     * @throws ApiException
     */
    public function callSample(): void
    {
        $config = Config::generateConfig();
        // Replace with the actual callback URI.
        $callbackUri = "https://www.samplesite.com";
        // Set the event type.
        $eventType = NotificationEventType::PRODUCT_STATUS_CHANGE;

        self::createNotificationSubscriptionSample($config, $callbackUri, $eventType);
    }
}

// Run the script.
$sample = new CreateNotificationSubscription();
$sample->callSample();
// [END merchantapi_create_notification_subscription]

File: python/examples/notifications/v1/create_notification_subscription_sample.py
# -*- coding: utf-8 -*-
# Copyright 2025 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""A module to create NotificationSubscription."""

# [START merchantapi_create_notification_subscription]
from examples.authentication import configuration
from examples.authentication import generate_user_credentials
from google.shopping.merchant_notifications_v1 import CreateNotificationSubscriptionRequest
from google.shopping.merchant_notifications_v1 import NotificationsApiServiceClient
from google.shopping.merchant_notifications_v1 import NotificationSubscription

_ACCOUNT = configuration.Configuration().read_merchant_info()
_PARENT = f"accounts/{_ACCOUNT}"


def create_notification_subscription():
  """Creates a Notification Subscription for a given Merchant Center account."""

  # Gets OAuth Credentials.
  credentials = generate_user_credentials.main()

  # Creates a client.
  client = NotificationsApiServiceClient(credentials=credentials)

  # The callback URI to be used to push the notification to.
  callback_uri = "https://www.samplesite.com"

  # The event type to register for notifications.
  event_type = (
      NotificationSubscription.NotificationEventType.PRODUCT_STATUS_CHANGE
  )

  # Creates the NotificationSubscription object.
  notification_subscription = NotificationSubscription()
  # Uncomment the next line to create a notification subscription for all
  # managed accounts.
  # notification_subscription.all_managed_accounts = True
  notification_subscription.target_account = _PARENT
  notification_subscription.call_back_uri = callback_uri
  notification_subscription.registered_event = event_type

  # Creates the request object.
  request = CreateNotificationSubscriptionRequest(
      parent=_PARENT, notification_subscription=notification_subscription
  )

  print("Sending create Notification Subscription request:")

  # Makes the API call.
  try:
    response = client.create_notification_subscription(request=request)
    print("Created Notification Subscription below:")
    print(response)
  except RuntimeError as e:
    print(e)


if __name__ == "__main__":
  create_notification_subscription()
# [END merchantapi_create_notification_subscription]
